pkg/cli/pod/pod.go

```diff
@@ -0,0 +1,390 @@
+	config, err := util.BuildConfig(listPodFlags.Master, listPodFlags.Kubeconfig)
+	if err != nil {
+		return err
+	}
+	if listPodFlags.allNamespace {
+		listPodFlags.Namespace = ""
+	}
+
+	labelSelector, err := createLabelSelector(listPodFlags.JobName, listPodFlags.QueueName)
+	if err != nil {
+		return err
+	}
+
+	client := kubeclientset.NewForConfigOrDie(config)
+	pods, err := client.CoreV1().Pods(listPodFlags.Namespace).List(ctx, metav1.ListOptions{
+		LabelSelector: labelSelector.String(),
+	})
+	if err != nil {
+		return err
+	}
+
+	if len(pods.Items) == 0 {
+		fmt.Printf("No resources found\n")
+		return nil
+	}
+	PrintPods(pods, os.Stdout)
+
+	return nil
+}
+
+func PrintPods(pods *corev1.PodList, writer io.Writer) {
+	maxNameLen := 0
+	maxReadyLen := 0
+	maxStatusLen := 0
+	maxRestartLen := 0
+	maxAgeLen := 0
+
+	var infoList []PodInfo
+	for _, pod := range pods.Items {
+		info := printPod(&pod)
+		infoList = append(infoList, info)
+		// update max length for each column
+		if len(info.Name) > maxNameLen {
+			maxNameLen = len(info.Name)
+		}
+		if len(info.ReadyContainers) > maxReadyLen {
+			maxReadyLen = len(info.ReadyContainers)
+		}
+		if len(info.Status) > maxStatusLen {
+			maxStatusLen = len(info.Status)
+		}
+		if len(info.Restarts) > maxRestartLen {
+			maxRestartLen = len(info.Restarts)
+		}
+		if len(info.CreationTimestamp) > maxAgeLen {
+			maxAgeLen = len(info.CreationTimestamp)
+		}
+	}
+	columnSpacing := 8
+	maxNameLen += columnSpacing
+	maxReadyLen += columnSpacing
+	maxStatusLen += columnSpacing
+	maxRestartLen += columnSpacing
+	maxAgeLen += columnSpacing
+	formatStr := fmt.Sprintf("%%-%ds%%-%ds%%-%ds%%-%ds%%-%ds\n", maxNameLen, maxReadyLen, maxStatusLen, maxRestartLen, maxAgeLen)
+	_, err := fmt.Fprintf(writer, formatStr, Name, Ready, Status, Restart, Age)
+	if err != nil {
+		fmt.Printf("Failed to print Pod information: %s.\n", err)
+		return
+	}
+	for _, info := range infoList {
+		_, err := fmt.Fprintf(writer, formatStr, info.Name, info.ReadyContainers, info.Status, info.Restarts, info.CreationTimestamp)
+		if err != nil {
+			fmt.Printf("Failed to print Pod information: %s.\n", err)
+			return
+		}
+	}
+}
+
+// createLabelSelector creates a label selector based on the provided job name or queue name.
+func createLabelSelector(jobName, queueName string) (labels.Selector, error) {
+	var labelSelector labels.Selector
+	reqs := []labels.Requirement{}
+
+	// Add label requirements based on jobName label
+	if jobName != "" {
+		inRequirement, err := labels.NewRequirement(v1alpha1.JobNameKey, selection.In, []string{jobName})
+		if err != nil {
+			return nil, err
+		}
+		reqs = append(reqs, *inRequirement)
+	}
+	// Add label requirements based on queueName label
+	if queueName != "" {
+		inRequirement, err := labels.NewRequirement(v1alpha1.QueueNameKey, selection.In, []string{queueName})
+		if err != nil {
+			return nil, err
+		}
+		reqs = append(reqs, *inRequirement)
+	}
+
+	if len(reqs) > 0 {
```

Comment:
If both label and queue are not provided, it will print all pods under all vcjobs,
but if just provide queue, it will print all pods under a queue, are these pods under vcjob? because seems that there is no vc job lable.

Reply:
According to the current implementation, all pods implemented using queue will be printed, not just vcjob.

Reply:
Do we only need the vcjob pod? I understand that the queue management pod should also need to be printed

Reply:
For vcjob，pods under it will add a label `volcano.sh/queue-name` to specify, which is your current implement covered, for non vcjob, there is no such  a label, user can specify annotation `scheduling.volcano.sh/queue-name` to set a queue name for bare pod，which is not covered in your current code. I think we should also  consider this.

Reply:
I added both labels to the search
```go
          // Add label requirements based on queueName label
	if queueName != "" {
		// pods under vcjobs
		inRequirementForVcJob, err := labels.NewRequirement(v1alpha1.QueueNameKey, selection.In, []string{queueName})
		if err != nil {
			return nil, err
		}
		// pods under specific queue
		inRequirementForPod, err := labels.NewRequirement(schedulingv1beta1.QueueNameAnnotationKey, selection.In, []string{queueName})
		if err != nil {
			return nil, err
		}
		reqs = append(reqs, *inRequirementForVcJob)
		reqs = append(reqs, *inRequirementForPod)
	}
```